### PR TITLE
Integrated Frontend code and updated one-click.sh

### PR DESCRIPTION
**Pull request by dr-natetorious**
*Issue #, if available:*

This change adds an [AWS Amplify](https://aws.amazon.com/amplify/) based web interface for testing out the sample. You can deploy it by running the `./one-click.sh` script

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: one-click.sh [hunk 1]
```diff
@@ -31,8 +31,8 @@ color_reset
 if [ -z "$TOTAL_COLLECTIONS" ]
 then
   color_reset
-  export TOTAL_COLLECTIONS=10
-  echo "TOTAL_COLLECTIONS not set, defaulted to 10 collections (200M faces)."
+  export TOTAL_COLLECTIONS=1
+  echo "TOTAL_COLLECTIONS not set, defaulted to 1 collections (200M faces)."
```
Comment (dr-natetorious) [inline one-click.sh:35]: 1 collection is 20M faces.

File: one-click.sh [hunk 2]
```diff
@@ -390,6 +390,72 @@ color_reset
...
+        sleep 1
+        timer=$((timer + 1))
+    done
+
+    curl $(echo ${job} | jq -r '.job.steps[] | select ( .stepName  | contains ( "DEPLOY")) | .logUrl')
+
+}
+
+
+echo "###########################################################"
```
Comment (dr-natetorious) [inline one-click.sh:416]: Missing color tags see above examples
```
color_green
echo "##########################"
echo "Section Name"
echo "###########################"
color_reset

color_red
echo "Error: "
color_reset
```

File: one-click.sh [hunk 3]
```diff
@@ -390,6 +390,72 @@ color_reset
...
+echo "# Build react   APP"
+echo "###########################################################"
+_pwd=$(pwd)
```
Comment (dr-natetorious) [inline one-click.sh:419]: This should be `$BASE_DIR` and all paths use it e.g., `cd $BASE_DIR/amplify/src/frontend`

File: one-click.sh [hunk 4]
```diff
@@ -390,6 +390,72 @@ color_reset
...
+aws amplify create-branch --app-id ${APP_ID} --environment-variables "$data" --branch-name ${BRANCH_NAME} 2>&1 >/dev/null || true
+
+echo "###########################################################"
+echo "#  Create ZIP"
+echo "###########################################################"
+ echo Creating zip and uploading to S3 bucket ${S3_ASSET_BUCKET}
+    _pwd=$(pwd)
+    echo $PWD
+    cd 'build'
+    rm ${BRANCH_NAME}.zip 2>/dev/null || true
```
Comment (dr-natetorious) [inline one-click.sh:444]: reduces to `rm -f ${BRANCH_NAME}.zip`

File: one-click.sh [hunk 5]
```diff
@@ -390,6 +390,72 @@ color_reset
...
+echo "#  Create Brach"
+echo "###########################################################"
+
+aws amplify create-branch --app-id ${APP_ID} --environment-variables "$data" --branch-name ${BRANCH_NAME} 2>&1 >/dev/null || true
+
+echo "###########################################################"
+echo "#  Create ZIP"
+echo "###########################################################"
+ echo Creating zip and uploading to S3 bucket ${S3_ASSET_BUCKET}
+    _pwd=$(pwd)
```
Comment (dr-natetorious) [inline one-click.sh:441]: You can improve readability in this section with `pushd path` and `popd` commands

File: src/frontend/public/index.html [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (dr-natetorious) [inline src/frontend/public/index.html:27]: Set the title to `Identity Verification using Amazon Rekognition`. Include a link to [the blog](https://aws.amazon.com/blogs/machine-learning/identity-verification-using-amazon-rekognition/) on the home page.

File: src/frontend/src/Components/FileUpload.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
+import React, { useRef, useState } from "react";
+import styled from "styled-components";
+import IconButton from '@mui/material/IconButton';
+import FileUploadIcon from '@mui/icons-material/FileUpload';
+
```
Comment (dr-natetorious) [inline src/frontend/src/Components/FileUpload.js:5]: Remove excess whitespace

File: src/frontend/src/Components/FileUpload.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+  right: 0;
+  bottom: 0;
+  opacity: 0;
+
+  &:focus {
+    outline: none;
+  }
+`;
+
+
```
Comment (dr-natetorious) [inline src/frontend/src/Components/FileUpload.js:38]: Same comment for remainder of file. If you need to place a visual separator use `/** */` style comments above elements.

File: src/frontend/src/Components/Form.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+import StyledButton from '../Components/ButtonTheme';
+import Stack from "@mui/material/Stack";
+import { API } from "aws-amplify";
+import Typography from '@mui/material/Typography';
+import { Navigate } from "react-router-dom";
+import Box from '@mui/material/Box';
+import CircularProgress from '@mui/material/CircularProgress';
+
+
+const Form = ({ imgSrc }) => {
```
Comment (dr-natetorious) [inline src/frontend/src/Components/Form.js:12]: Can you add some block comments that describe what is happening in the critical sections of the code?

File: src/frontend/src/Components/Form.js [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+                let responseData = response;
+                if (responseData.status === "SUCCEEDED") {
+                    let responseSuccessData = JSON.parse(responseData.output)
+                    setregisterSuccess({ "userName": responseSuccessData.UserId, "imageId": responseSuccessData.ImageId, "label": "       Successfully Registered User", "fName": fName, "lName": lName, "age": age})
+                } else {
+                    setError(responseData.status + " : " + responseData.error)
+                }
+                setprogress(false)
+                
+            })
```
Comment (dr-natetorious) [inline src/frontend/src/Components/Form.js:48]: Please format this section of the document

File: src/frontend/src/Components/WebCam.jsx [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+import StyledButton from './ButtonTheme';
+
+
+const videoConstraints = {
+  width: 1300,
+  height: 720,
+  facingMode: "user"
+  
+};
+
```
Comment (dr-natetorious) [inline src/frontend/src/Components/WebCam.jsx:13]: Cleanup and document this class. It looks like the more resuable aspects of the project.

File: src/frontend/src/Pages/Profile.jsx [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+}));
+
+const initialUserState = {
+    fName: "Amit",
+    lName: "Gupta",
+    id: "",
+    age: "1970-01-01",
+};
+
+export default function RegidterWithId() {
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Profile.jsx:37]: Spelling `RegisterWithIdCard`

File: src/frontend/src/Pages/Profile.jsx [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+        localStorage.setItem("pic", image);
+        setImgSrc(image);
+    }
+
+    const recapture = () => {
+        localStorage.removeItem("pic")
+        setImgSrc(null);
+    }
+
+    const update = () => {
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Profile.jsx:60]: Remove excess spacing. If you need a visual separator use `/** */` style comments.

File: src/frontend/src/Pages/Profile.jsx [hunk 14]
```diff
@@ -0,0 +1,244 @@
...
+                if (responseData.status === "SUCCEEDED") {
+                    let responseSuccessData = JSON.parse(responseData.output)
+                    setregisterSuccess({ "userName": responseSuccessData.UserId, "imageId": responseSuccessData.ImageId, "label": "       Successfully Registered User", "fName": properties.fName, "lName": properties.lName, "age": properties.age })
+
+                } else {
+                    console.log(responseData.error)
+                    setError(responseData.status + " : Identification information doesn't match")
+                }
+                setprogress(false)
+            })
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Profile.jsx:93]: Cleanup indenting for readability

File: src/frontend/src/Pages/Profile.jsx [hunk 15]
```diff
@@ -0,0 +1,244 @@
...
+        // console.log(userDetails)
+        const id = [event.target.id]
+        properties[id] = event.target.value
+        setProperties(properties)
+        console.log(properties)
+    };
+
+
+    return (
+        <>
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Profile.jsx:114]: Is `<>` a valid tag?

File: src/frontend/src/Pages/Profile.jsx [hunk 16]
```diff
@@ -0,0 +1,244 @@
...
+                    direction="row"
+                    justifyContent="center"
+                    alignItems="stretch"
+                  >
+                            {imgSrc ?
+                            <>
+                              <Grid item xs={5} md={5}
+                                >
+                                <Item elevation={0}
+
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Profile.jsx:128]: Use standard indenting to improve readability

File: src/frontend/src/Pages/Register.js [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+import Title from '../Components/PageTitle'
+import StyledButton from '../Components/ButtonTheme';
+
+
+const Item = styled(Paper)(({ theme }) => ({
+    ...theme.typography.body2,
+    textAlign: 'center',
+    color: theme.palette.text.secondary,
+}));
+
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/Register.js:17]: Same comments about excess whitespace, formatting, and "some" documents

File: src/frontend/src/Pages/RegisterWithID.js [hunk 18]
```diff
@@ -0,0 +1,238 @@
...
+}));
+
+const initialUserState = {
+    fName: "",
+    lName: "",
+    id: "",
+    age: "",
+};
+
+export default function RegidterWithId() {
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/RegisterWithID.js:38]: Spelling `RegisterWithIdCard`

Comment (dr-natetorious) [reply src/frontend/src/Pages/RegisterWithID.js:38]: How does this differ from the version in the Profile page?

File: src/frontend/src/Pages/SignIn.jsx [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+    const capture = (idcard) => {
+
+        setImgSrc(idcard);
+        makeAuthCall()
+
+    }
+
+
+    return (
+        <>
```
Comment (dr-natetorious) [inline src/frontend/src/Pages/SignIn.jsx:93]: Same comments about formatting, whitespace, and readability

**Review by dr-natetorious — COMMENTED** [review]
Thanks for adding the Frontend UI. Can you do a once-over and cleanup [the formatting of each file](https://stackoverflow.com/questions/43666270/how-do-i-format-all-files-in-a-visual-studio-code-project)?  

It would also be useful to include some documentation as a `README.md` and with inline `/** */`-style comments. You do not need to go overboard, but even a little goes a long way. In the README make sure to specify:

1. How do I build the amplify app
2. How do I run the amplify app
3. How is the code organized and any key points

File: src/frontend/src/Pages/Profile.jsx (hunk 15 above)
Comment (vineetkachhawaha) [reply src/frontend/src/Pages/Profile.jsx:107]: Yes. Its a empty tag for declaring fragments

**Pull request closed without merging**